plaid/src/executor/mod.rs

```diff
@@ -316,8 +318,58 @@ fn execution_loop(
             Some(module) => module,
         };
 
+        // If ANY of the modules that operate on this log type cannot be executed in parallel, we need to check
+        // the rules in flight list to see if a rule is already being executed.
+        if execution_modules
+            .iter()
+            .any(|module| !module.parallel_execution_enabled)
+        {
+            let strif = match single_threaded_rules_in_flight.read() {
+                Ok(strif) => strif,
+                Err(e) => {
+                    // .read() will return an error if the lock is poisoned (which should never happen). If this occurs,
+                    // we'll clear the poison and send the message back into the execution queue
+
+                    error!("Failed to acquire read lock. Error: {e}");
+                    single_threaded_rules_in_flight.clear_poison();
+
+                    if let Err(e) = sender.send(message.clone()) {
+                        error!("Failed to send message back into execution queue. Error: {e}");
+                    }
+                    continue;
+                }
+            };
+
+            // If this rule is currently being processed, send the message back into the execution queue
+            // This allows us to guarantee that the rule will only be executed on 1 thread at any given time
+            if execution_modules
+                .iter()
+                .any(|module| strif.contains(&module.name))
+            {
+                if let Err(e) = sender.send(message.clone()) {
+                    error!("Failed to send message back into execution queue. Error: {e}");
```

Comment:
This should mention that the log is now lost as well

Reply:
There should also be an info or trace log here that a log was delayed due to parallel execution prevention.